app/assets/javascripts/covid19/src/data/repositories/LigandsApiRepository.ts

```diff
@@ -25,16 +25,21 @@ export class LigandsApiRepository implements LigandsRepository {
                 (ligandToImageData): FutureData<LigandImageData> => {
                     if (!ligandToImageData)
                         return Future.error(err("Error: the api response is undefined."));
-                    if (_.has(ligandToImageData, "detail"))
+                    else if (_.has(ligandToImageData, "detail"))
                         return Future.error(err('Error: "detail": Not found.'));
+
                     const data = ligandToImageData as LigandToImageData;
```

Comment:
Typically, we shouldn't need this kind of casting. I see that `ligandTomImageData` is a union, and the prop `detail` can be used as a discriminator for the error/success branches. TS does not understand `_.has` to narrow the object type, but the predicate `key in object` does work. 


Proposal: else if ("detail" in ligandToImageData)`. Now, `as LigandToImageData` should be innecessary.

Reply:
I "[...] will never go to bed without learning one more thing"